components/authz-service/server/v2/project_update_manager.go

```diff
@@ -523,3 +524,233 @@ func getEstimatedTimeCompleteInSec(event *automate_event.EventMsg) (time.Time, e
 	return time.Time{}, fmt.Errorf("Event message sent without a %s field eventID: %q",
 		fieldName, event.EventID)
 }
+
+func NewWorkflowExecutor(domainServices []string) *multiworkflow.MultiWorkflow {
+	workflowMap := make(map[string]cereal.WorkflowExecutor, len(domainServices))
+	for _, d := range domainServices {
+		workflowMap[d] = project_update_tags.NewWorkflowExecutorForDomainService(d)
+	}
+	return multiworkflow.NewMultiWorkflowExecutor(workflowMap)
+}
+
+type CerealProjectUpdateManager struct {
+	manager        *cereal.Manager
+	workflowName   string
+	instanceName   string
+	domainServices []string
+}
+
+func RegisterCerealProjectUpdateManager(manager *cereal.Manager, workflowName string, instanceName string, domainServices []string) (ProjectUpdateMgr, error) {
+	domainServicesWorkflowExecutor := NewWorkflowExecutor(domainServices)
+
+	if err := manager.RegisterWorkflowExecutor(workflowName, domainServicesWorkflowExecutor); err != nil {
+		return nil, err
+	}
+
+	updateManager := &CerealProjectUpdateManager{
+		manager:        manager,
+		workflowName:   workflowName,
+		instanceName:   instanceName,
+		domainServices: domainServices,
+	}
+
+	return updateManager, nil
+}
+
+func (m *CerealProjectUpdateManager) Cancel() error {
+	err := m.manager.CancelWorkflow(context.Background(), m.workflowName, m.instanceName)
+	if err == cereal.ErrWorkflowInstanceNotFound {
+		return nil
+	}
+	return err
+}
+
+func (m *CerealProjectUpdateManager) Start() error {
+	params := map[string]interface{}{}
+
+	for _, svc := range m.domainServices {
+		params[svc] = project_update_tags.DomainProjectUpdateWorkflowParameters{
+			ProjectUpdateID: "project-update-id",
```

Comment:
Do we need the `ProjectUpdateID` anymore?

Reply:
Up to you. I can wire one up if it's helpful

Reply:
I don't think we would need it anymore because the ProjectUpdateID was used before to ensure what update the cancel, start, and status requests were talking about. This was because with using the event-service old requests for prior project updates could have appeared during the next project update. 

If I am thinking of how the cereal service is working correctly, then that coordination problem could not happen.

Reply:
Ya, all the work that happens is getting tied to 1 workflow id